Tests/AblyChatTests/ChatAPITests.swift

```diff
@@ -52,6 +52,52 @@ struct ChatAPITests {
         #expect(message == expectedMessage)
     }
 
+    @Test
+    func sendMessage_includesHeadersInBody() async throws {
+        // Given
+        let realtime = MockRealtime.create {
+            (MockHTTPPaginatedResponse.successSendMessage, nil)
+        }
+        let chatAPI = ChatAPI(realtime: realtime)
+
+        // When
+        _ = try await chatAPI.sendMessage(
+            roomId: "", // arbitrary
+            params: .init(
+                text: "", // arbitrary
+                // The exact value here is arbitrary, just want to check it gets serialized
+                headers: ["numberKey": .number(10), "stringKey": .string("hello")]
```

Comment:
Why can't it be ["numberKey": 10, "stringKey": "hello"], where dictionary is [`String`, `AnyPrimitiveType`], where `AnyPrimitiveType` is just an empty protocol, to which `String`, `Int` etc are conformed?

Reply:
ok, this will not work for nested dictionaries